### ISSUE TITLE
Proxy/meta2 : container auto-created but not initialized

### ISSUE THREAD

**Issue by Reaskel** [open-io/oio-sds#80]
When a reference is linked for the first time with a service, a corresponding container is automatically created but not initialized.
This is confusing because a lot of put/get/delete meta2 functions will consider that ghost container as existing and will bring unexpected results.

The two solutions could be either initializing those auto-created containers or restrict container creation to the put method only.

**Comment by Reaskel** [comment]
Thanks @jfsmig, your last update #103 allowed several meta2 functions to handle bad configurations.

However, there remains some functions which still consider linked references as containers :

> HEAD on container
SET / GET / DEL on container properties
SET on container's storage policy

**Comment by Reaskel** [comment]
Here is an example of request sequence to get the error : 

First, create a reference : 
>curl -XPUT 'IP:PORT/v2.0/dir/$NS/$ACCOUNT/$REFNAME'

Then, link the reference with an existing service : 
>curl -XPOST 'IP:PORT/v2.0/dir/$NS/$ACCOUNT/$REFNAME/meta2/action'
with the following json : {"action": "Link", "args": None } 

Finally, if you try a GET on the non-existent container corresponding to your linked reference, you'll get results, instead of a 404 error : 
>curl -XGET 'IP:PORT/v2.0/m2/$NS/$ACCOUNT/$REFNAME'

**Comment by jfsmig** [comment]
All the "meta2" requests handlers perform a check on the container's file. That check is responsible for the absence of automatic creation. Let's check if a fix is welcome...

**Comment by jkasarherou** [comment]
There is still some unresolved issues for this, for example for a container/get_properties, the output does not indicate that the actual container is created (response 200 with body):

```    
> POST /v3.0/NS/container/get_properties?acct=test_account&ref=h HTTP/1.1
> Host: 127.0.0.1:6002
> User-Agent: curl/7.44.0-DEV
> Accept: */*
>
< HTTP/1.1 200 OK
< Server: metacd_http/0.7.0
< Connection: Close
< Content-Type: application/json
< Content-Length: 316
< Transfer-Encoding: identity
<
* Closing connection 0
{"schema_version":"1.8","sys.name":"3B63138AA5FF9CC6D0CA536E51115A35AEBF344699CB4C5692F5E644DB4D0FB7.1","sys.type":"meta2","version:main.admin":"1:0","version:main.alias_v2":"1:0","version:main.chunk_v2":"1:0","version:main.content_header_v2":"1:0","version:main.content_v2":"1:0","version:main.properties_v2":"1:0"}% 
```

**Comment by jfsmig** [comment]
@Reaskel : This issue is 4 months old, and could not be reproduced. Can you please provide a minimal script to reproduce the errors, of course adapted to the latest v3 URL scheme ? thx

**Comment by jfsmig** [comment]
Fixed at commit e61c18ffeba95fbcf4549184c74c97bcaf78c13c
